libs/core/algorithms/tests/performance/benchmark_reduce_deterministic.cpp

```diff
@@ -0,0 +1,224 @@
+//  Copyright (c) 2024 Shreyas Atre
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/config.hpp>
+#include <cstddef>
+
+#if !defined(HPX_COMPUTE_DEVICE_CODE)
+#include <hpx/algorithm.hpp>
+#include <hpx/chrono.hpp>
+#include <hpx/execution.hpp>
+#include <hpx/init.hpp>
+#include <hpx/modules/testing.hpp>
+#include <hpx/parallel/algorithms/reduce.hpp>
+#include <hpx/parallel/algorithms/reduce_deterministic.hpp>
+
+#include <iostream>
+#include <numeric>
+#include <random>
+#include <vector>
+
+int seed = 1000;
+std::mt19937 gen(seed);
+
+template <typename T>
+T get_rand(T LO = (std::numeric_limits<T>::min)(),
+    T HI = (std::numeric_limits<T>::max)())
+{
+    return LO +
+        static_cast<T>(std::rand()) /
+        (static_cast<T>(static_cast<T>((RAND_MAX)) / (HI - LO)));
+}
+
+///////////////////////////////////////////////////////////////////////////////
+template <typename PolicyT, typename IteratorT, typename InitVal, typename Op>
+void bench_reduce_deterministic(const PolicyT& policy,
+    const IteratorT& deterministic_shuffled, const InitVal& val_det,
+    const Op& op)
+{
+    // check if different type for deterministic and nondeeterministic
+    // and same result
+
+    auto r1_shuffled = hpx::experimental::reduce_deterministic(policy,
+        std::begin(deterministic_shuffled), std::end(deterministic_shuffled),
+        val_det, op);
+
+    HPX_UNUSED(r1_shuffled);
```

Comment:
We started to move away from `HPX_UNUSED`, please use `[[maybe_unused]]` wherever possible.